### PR TITLE
Feature/recent refs

### PR DESCRIPTION
**Pull request by lpoli**
This PR contains test for feature that lists recently added refs.

### CODE REVIEW

File: tests/cli_tests/zboxcli_recent_refs_test.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+	Offset int    `json:"offset"`
+	Refs   []ORef `json:"refs"`
+}
+
+type ORef struct {
+	SimilarField
+	ID int64 `json:"id"`
+}
+
+type SimilarField struct {
```
Comment (stewartie4) [inline tests/cli_tests/zboxcli_recent_refs_test.go:27]: can we move these models out of the test file please?

File: internal/cli/util/utils.go [hunk 2]
```diff
@@ -210,3 +212,20 @@ func Contains(slice []string, val string) (int, bool) {
 	}
 	return -1, false
 }
+
+// GetSubPaths will return [/ /a /a/b /a/b/c /a/b/c/d /a/b/c/d/e] for
+// path /a/b/c/d/e
+func GetSubPaths(p string) (paths []string, err error) {
```
Comment (stewartie4) [inline internal/cli/util/utils.go:218]: handy

File: tests/cli_tests/zboxcli_recent_refs_test.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+func TestRecentlyAddedRefs(t *testing.T) {
+	t.Parallel()
+
+	// Create a folder to keep all the generated files to be uploaded
+	err := os.MkdirAll("tmp", os.ModePerm)
+	require.Nil(t, err)
+
+	t.Run("Recently Added Refs Should be listed", func(t *testing.T) {
+		allocationID := setupAllocation(t, configPath, map[string]interface{}{
```
Comment (stewartie4) [inline tests/cli_tests/zboxcli_recent_refs_test.go:54]: can these tests run in parallel?

File: tests/cli_tests/zboxcli_recent_refs_test.go [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+		err = json.Unmarshal([]byte(output[len(output)-1]), &result)
+		require.Nil(t, err)
+
+		require.Nil(t, err)
+		require.Equal(t, 0, result.Offset)
+		require.Len(t, result.Refs, 0)
+
+	})
+
+	t.Run("Invalid parameters should return error", func(t *testing.T) {
```
Comment (stewartie4) [inline tests/cli_tests/zboxcli_recent_refs_test.go:151]: what about attempting to list someone else's allocation? or a ref that doesn't exist

**Review by stewartie4 — CHANGES_REQUESTED** [review]
minor comments, thanks for the PR

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | feature/recent-refs  |
| gosdk  | feature/list-recent-files  |
| zboxcli  | feature/list-recent-files  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: go.mod [hunk 5]
```diff
@@ -3,16 +3,14 @@ module github.com/0chain/system_test
 go 1.16
 
 require (
+	github.com/0chain/gosdk v1.8.5
```
Comment (stewartie4) [inline go.mod:6]: was go mod tidy run necessary? I wonder how we didn't notice

File: tests/cli_tests/zboxcli_recent_refs_test.go (hunk 4 above)
Comment (lpoli) [reply tests/cli_tests/zboxcli_recent_refs_test.go:151]: It is redundant to test for `refs that does not exist.`
We can add to test for some else's allocation recent file listing.

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | feature/recent-refs  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | fix/limit-access-on-recent-refs  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Review by stewartie4 — APPROVED** [review]

**Pull request merged**